### ISSUE TITLE
Nested block types?

### ISSUE THREAD

**Issue by alerque** [alerque/vim-commonmark#18]
An interesting edge case the information we're currently collecting doesn't seem to let us draw correctly:

```markdown
> foo
>
> * bar
> * baz
>
> qix
```

![image](https://user-images.githubusercontent.com/173595/82257153-3b5bbf00-9960-11ea-8c34-9a86179ee2a7.png)

Not only am I not sure what's going on here (I haven't debugged it, only saving this to check back with later) but I'm not ever sure what I think *should* happen in this case.

**Comment by fmoralesc** [comment]
And now we come back to the issue of introspectability :p

**Comment by fmoralesc** [comment]
Well, I think neither blockquotes nor lists should be highlighted, so imho nothing should happen.

**Comment by alerque** [comment]
Personally, I would only highlight the delimiters (`> `, `* `) except that it's actually pretty helpful to see when subsequent paragraphs in lists are parsed as part of the list item or not. In that case it's really useful to have some highlighting. Also block quote are a lot easier to scan for scrolling through a large document if they are visible as blocks. So I'm torn.

In any case we need to both allow this as an option if people want it (you can `hi link cmarkBlockquote NONE` to your heart's content) but also we need the ability to know what block level item we are in even if we don't give it a color!

**Comment by fmoralesc** [comment]
I agree. I think the issue here is that the embedded text is 'boxed in', and the offsets don't take that into account, so there's 'bleed' of some sort.
